Define custom platform for HSM tests

heaths opened the pull request:
Resolves #16483

heaths committed:
Define custom platform for HSM tests

Resolves #16483

heaths commented:
/azp run net - keyvault - tests

benbp commented on sdk/keyvault/tests.yml:
```diff
@@ -9,3 +9,11 @@ extends:
+      Linux_HSM:
+        OSVmImage: "ubuntu-18.04"
+        TestTargetFramework: netcoreapp2.1
+        ArmTemplateParameters: "@{enableHsm = $true}"
```
I think you'll need to add one more platform variable:

```
SupportedClouds: 'Public,UsGov'
```

It looks like HSM is not yet available in china and canary regions: [availability](https://azure.microsoft.com/en-us/global-infrastructure/services/?regions=china-non-regional,china-east,china-east-2,china-north,china-north-2,us-east,us-west,us-west-2,non-regional,usgov-non-regional,us-dod-central,us-dod-east,usgov-arizona,usgov-texas,usgov-virginia,us-central,us-east-2,us-north-central,us-south-central,us-west-central&products=azure-dedicated-hsm)

![image](https://user-images.githubusercontent.com/1020379/104501258-77a5f480-55ad-11eb-81d5-84d91d626a63.png)

heaths replied:
It's in canary, but has a different set of regions, prefixes, and other goop. The two clouds should suffice, though I wonder - since it's just compat testing our KeyClient with Managed HSM (APIs between KV and HSM are designed to be the same), do we need another cloud? HSMs are pretty pricey and limited allocations make them potentially problematic.

heaths replied:
I think I see now: without this, the pipeline will try all clouds. This limits it. Change made.

benbp replied:
Yeah, I think it makes sense to just test public here too, given what you say about the intent of the test. In that case just `SupportedClouds: Public`

heaths committed:
Resolve feedback, fix other recent test issues

heaths commented:
/azp run net - keyvault - tests

heaths commented:
I'm going to have to put this on hold. There are some service issues preventing tests from running on Managed HSM.

heaths committed:
Resolve a few live test issues

heaths committed:
Update service version to 7.2

heaths committed:
Do not log x-ms-request-id by default

This will only be logged for tests for diagnostic purposes.

heaths committed:
Update service version to 7.2 and re-record tests

Resolves #16483, though #18501 is tracking enabling the custom platform now once Managed HSM service issues are resolved.

heaths committed:
Update public APIs

Also updated the autorest commit for 7.2-preview to the last that touched the 7.2-preview swagger folder.

heaths committed:
Fix test issues

christothes commented on sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/SecretClientLiveTests.cs:
```diff
@@ -53,7 +53,7 @@ public async Task SetSecret()
         {
             string secretName = Recording.GenerateId();
-            IResolveConstraint createdUpdatedConstraint = Is.EqualTo(DateTimeOffset.FromUnixTimeSeconds(1596061817));
+            IResolveConstraint createdUpdatedConstraint = Is.EqualTo(DateTimeOffset.FromUnixTimeSeconds(1612561237));
```
what's this magic number? :)

heaths replied:
Lotto numbers. Or the current time from the first recording, which I then updated in the subsequent recording (in this case, sync ran first, then async, so I updated async manually). Probably should either stop asserting this (arguably going too far asserting it) or figure out a better way. I only need to do this when re-recording, which for secrets is only when new service versions roll out.

heaths commented:
/azp run net - keyvault - tests
